sources/kms/key-policy.go

```diff
@@ -0,0 +1,104 @@
+package kms
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+type keyPolicyClient interface {
+	GetKeyPolicy(ctx context.Context, params *kms.GetKeyPolicyInput, optFns ...func(*kms.Options)) (*kms.GetKeyPolicyOutput, error)
+	ListKeyPolicies(ctx context.Context, params *kms.ListKeyPoliciesInput, optFns ...func(*kms.Options)) (*kms.ListKeyPoliciesOutput, error)
+}
+
+func getKeyPolicyFunc(ctx context.Context, client keyPolicyClient, scope string, input *kms.GetKeyPolicyInput) (*sdp.Item, error) {
+	output, err := client.GetKeyPolicy(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.Policy == nil {
```

Comment:
It would be cool if we could actually parse the policy here. Can we use the same approach is we did here? https://github.com/overmindtech/aws-source/blob/5b480dbf13ece1d7fff94401217e642f1ecc2f3a/sources/ec2/vpc_endpoint.go#L27-L50